service/src/main/java/bio/terra/cbas/controllers/util/RunSetsHelper.java

```diff
@@ -0,0 +1,286 @@
+import static bio.terra.cbas.model.RunSetState.RUNNING;
+import static bio.terra.cbas.models.CbasRunStatus.INITIALIZING;
+import static bio.terra.cbas.models.CbasRunStatus.QUEUED;
+import static bio.terra.cbas.models.CbasRunStatus.SYSTEM_ERROR;
+
+import bio.terra.cbas.common.DateUtils;
+import bio.terra.cbas.common.exceptions.InputProcessingException;
+import bio.terra.cbas.config.CbasApiConfiguration;
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.dependencies.wds.WdsClientUtils;
+import bio.terra.cbas.dependencies.wds.WdsService;
+import bio.terra.cbas.dependencies.wds.WdsServiceApiException;
+import bio.terra.cbas.dependencies.wds.WdsServiceException;
+import bio.terra.cbas.dependencies.wes.CromwellService;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.model.RunSetState;
+import bio.terra.cbas.model.RunStateResponse;
+import bio.terra.cbas.models.CbasRunSetStatus;
+import bio.terra.cbas.models.CbasRunStatus;
+import bio.terra.cbas.models.Run;
+import bio.terra.cbas.models.RunSet;
+import bio.terra.cbas.runsets.inputs.InputGenerator;
+import bio.terra.cbas.runsets.types.CoercionException;
+import bio.terra.cbas.util.UuidSource;
+import bio.terra.common.iam.BearerToken;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.Lists;
+import cromwell.client.model.WorkflowIdAndStatus;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.databiosphere.workspacedata.model.RecordResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RunSetsHelper {
+
+  private final RunDao runDao;
+  private final RunSetDao runSetDao;
+  private final CromwellService cromwellService;
+  private final WdsService wdsService;
+  private final CbasApiConfiguration cbasApiConfiguration;
+  private final UuidSource uuidSource;
+
+  private final Logger logger = LoggerFactory.getLogger(RunSetsHelper.class);
+
+  public RunSetsHelper(
+      RunDao runDao,
+      RunSetDao runSetDao,
+      CromwellService cromwellService,
+      WdsService wdsService,
+      CbasApiConfiguration cbasApiConfiguration,
+      UuidSource uuidSource) {
+    this.runDao = runDao;
+    this.runSetDao = runSetDao;
+    this.cromwellService = cromwellService;
+    this.wdsService = wdsService;
+    this.cbasApiConfiguration = cbasApiConfiguration;
+    this.uuidSource = uuidSource;
+  }
+
+  private record WdsRecordResponseDetails(
+      ArrayList<RecordResponse> recordResponseList, Map<String, String> recordIdsWithError) {}
+
+  private record RunAndRecordDetails(UUID runId, RecordResponse recordResponse) {}
+
+  @Async
+  public void triggerWorkflowSubmission(
```

Comment:
Async method that will fetch records data from WDS and submit to Cromwell in a different thread as allocated by `getAsyncExecutor` from `AsyncConfig`.